guides/micronaut-data-access-mybatis/micronaut-data-access-mybatis.adoc

```diff
@@ -0,0 +1,207 @@
+to sql queries:
+
+source:genre/GenreMapper[]
+
+And the implementation:
+
+source:genre/GenreMapperImpl[]
+<1> Use `javax.inject.Singleton` to designate a class a a singleton.
+<2> Inject easily the `SqlSessionFactory` bean created by the `@Factory`.
+<3> Use _try-with-resources_ to automatically close the sql session.
+<4> Get MyBatis mapper implementation for the interface.
+<5> Execute the desired method using the mapper. This will trigger the sql query.
+<6> In a database write access, commit the transaction.
+
+
+Create an interface to define the high level operations exposed to the application:
+
+source:genre/GenreRepository[]
+
+And the implementation using `GenreMapper`:
+
+source:genre/GenreRepositoryImpl[]
+
+=== Controller
+
+include::{commondir}/common-validation.adoc[]
+
+Create two classes to encapsulate Save and Update operations:
+
+source:genre/GenreSaveCommand[]
+
+source:genre/GenreUpdateCommand[]
+
+Create a POJO to encapsulate Sorting and Pagination:
+
+source:ListingArguments[]
+
+Create a `ConfigurationProperties` object to encapsulate the configuration of the default `max` value.
+
+source:ApplicationConfiguration[]
+
+source:ApplicationConfigurationProperties[]
+
+Create `GenreController`, a controller which exposes a resource with the common CRUD operations:
+
+source:GenreController[]
+<1> The class is defined as a controller with the `@Controller` annotation mapped to the path `/genres`.
+<2> Constructor injection.
+<3> Maps a `GET` request to `/genres/{id}` which attempts to show a genre. This illustrates the use of a URL path variable.
+<4> Returning `null` when the genre doesn't exist makes Micronaut to response with 404 (not found).
+<5> Maps a `PUT` request to `/genres` which attempts to update a genre.
+<6> Add `@Valid` to any method parameter which requires validation. Use a POJO supplied as a JSON payload in the request to populate command.
+<7> It is easy to add custom headers to the response.
+<8> Maps a `GET` request to `/genres` which returns a list of genres. This mapping illustrates optional URL parameters.
+<9> Maps a `POST` request to `/genres` which attempts to save a genre.
+<10> Maps a `DELETE` request to `/genres/{id}` which attempts to remove a genre. This illustrates the use of a URL path variable.
+
+=== DB Schema
+
+Now that Mybatis is setup we need a way to create the database schema. For that we will use
+https://micronaut-projects.github.io/micronaut-flyway/latest/guide/index.html[Micronaut integration with
+Flyway].
+
+
+Add the next snippet to `build.gradle` to include the necessary dependencies:
+
+dependency:micronaut-flyway[groupId=io.micronaut.flyway]
+
+Configure the database migrations directory for Flyway in `application.yml`.
+
+resource:application.yml[tag=flyway]
+
+Create the file `V1__schema.sql` with the database schema creation:
+
+resource:databasemigrations/V1__schema.sql[]
```

Comment:
Please, change this and the app code to use the default location for the migrations. We changed it sometime ago in Flyway integration so it's good to have this guide aligned. See https://micronaut-projects.github.io/micronaut-flyway/latest/guide/#jpa